test/widgets/search_ride_page_test.dart

```diff
@@ -0,0 +1,967 @@
+        ];
+
+        whenRequest(processor).thenReturnJson(drives);
+        whenRequest(processor, urlMatcher: matches(RegExp('/rest/v1/drives.*id=eq.')))
+            .thenReturnJson(DriveFactory().generateFake().toJsonForApi());
+
+        await pumpMaterial(tester, const SearchRidePage());
+
+        final String start = faker.address.city();
+        await enterStartAndDestination(tester, start, faker.address.city());
+        await enterDateAndTime(tester, startTime);
+
+        verifyRequest(
+          processor,
+          urlMatcher: matches(RegExp('/rest/v1/drives.*start=eq\\.${RegExp.escape(Uri.encodeQueryComponent(start))}')),
+        );
+
+        expect(find.byType(RideCard, skipOffstage: false), findsNWidgets(2));
+
+        await tester.tap(find.byKey(const Key('searchRideWholeDayCheckbox')));
+        await tester.pump();
+        await tester.tap(find.byKey(const Key('searchRideAfterButton')));
+        await tester.pump();
+
+        expect(find.byType(RideCard, skipOffstage: false), findsOneWidget);
+
+        await tester.tap(find.byKey(const Key('searchRideAfterButton')));
+        await tester.pump();
+
+        expect(find.byType(RideCard, skipOffstage: false), findsNothing);
+      });
+
+      testWidgets('No results', (WidgetTester tester) async {
+        final List<Map<String, dynamic>> drives = [];
+
+        whenRequest(processor).thenReturnJson(drives);
+
+        await pumpMaterial(tester, const SearchRidePage());
+
+        await enterStartAndDestination(tester, faker.address.city(), faker.address.city());
+
+        expect(find.byType(RideCard, skipOffstage: false), findsNothing);
+        expect(find.byKey(const Key('searchRideNoResults')), findsOneWidget);
+      });
+
+      testWidgets('Too restrictive filters', (WidgetTester tester) async {
+        final DateTime startTime = DateTime.now();
+
+        final Profile driver = ProfileFactory().generateFake(profileFeatures: []);
+        final List<Map<String, dynamic>> drives = [
+          DriveFactory().generateFake(driver: NullableParameter(driver), startTime: startTime).toJsonForApi(),
+        ];
+
+        whenRequest(processor).thenReturnJson(drives);
+        whenRequest(processor, urlMatcher: matches(RegExp('/rest/v1/drives.*id=eq.')))
+            .thenReturnJson(DriveFactory().generateFake().toJsonForApi());
+
+        await pumpMaterial(tester, const SearchRidePage());
+
+        await enterStartAndDestination(tester, faker.address.city(), faker.address.city());
+        await enterFilter(tester, features: [Feature.values[random.integer(Feature.values.length)]]);
+
+        expect(find.byType(RideCard, skipOffstage: false), findsNothing);
+
+        final Finder relaxRestrictionsButton = find.byKey(const Key('searchRideRelaxRestrictions')).hitTestable();
+        await tester.scrollUntilVisible(relaxRestrictionsButton, 100, scrollable: find.byType(Scrollable).first);
```

Comment:
It should be noted that this is needed because the test screen is so small - on a normal screen the button would be visible, but here we have to scroll to it (@everyone)

Reply:
Yes, that sent me into a spin for quite some time because it appears the screen size is different if you execute tests one by one vs. if you execute the whole group because when I executed the test on its own, it worked fine every time, but within the group, it failed every time. So if you're experiencing anything like that, maybe that's your solution